backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java

```diff
@@ -1599,6 +1719,23 @@ void testConnectionCount() {
             ep.setEnabled(true);
             ep.setProperties(properties);
 
+            // Mock the Sources service calls.
+            final Secret secretTokenSecret = new Secret();
+            secretTokenSecret.id = new Random().nextLong(1, Long.MAX_VALUE);
+            secretTokenSecret.password = properties.getSecretToken();
```

Comment:
This and other changes from this PR are repeated several times. Could you please lower the amount of duplicated code?

Reply:
Thank you for taking care of this. I was finishing some work for the service accounts that got in the way and was planning to review this today.